server/polar/auth/dependencies.py

```diff
@@ -97,6 +97,11 @@ def __call__(self, auth_subject: AuthSubject[Subject]) -> AuthSubject[Subject]:
             else:
                 raise Unauthorized()
 
+        # Blocked subjects
+        blocked_at = getattr(auth_subject.subject, "blocked_at", None)
+        if blocked_at is not None:
+            raise Unauthorized()
```

Comment:
Correct, but also if subject is an organization for org scoped API calls for instance. 

Re: `403` that's fair. I'll oblige :) Nice to distinguish it in a pure API vs. 404. I'm a fan of the latter since it obfuscates the reason, but in this rare case I think it's OK to have it as `403`. Will update.